lib/livebook_web/live/output/frame_component.ex

```diff
@@ -26,31 +30,51 @@ defmodule LivebookWeb.Output.FrameComponent do
           assign(socket, outputs: outputs)
 
         :replace ->
-          socket
-          |> assign(outputs: outputs, empty: outputs == [])
-          |> update(:counter, &(&1 + 1))
+          prev_output_count = socket.assigns.output_count
+          output_count = length(outputs)
+          idx_map = append_idx_map(%{}, outputs)
+
+          socket = assign(socket, outputs: outputs, output_count: output_count, idx_map: idx_map)
+
+          # If there are less outputs we reset the counter to remove DOM elements
+          if output_count < prev_output_count do
+            update(socket, :counter, &(&1 + 1))
+          else
+            socket
+          end
 
         :append ->
           socket
-          |> assign(empty: false)
+          |> update(:idx_map, &append_idx_map(&1, outputs))
```

Comment:
We don't need this.

Reply:
So then the map has only as many entries as outputs rendered on replace. With the current API it's always 1, but if we add `Frame.render([...])`, then it's the number of rendered outputs.